src/cdtpComponents/cdtpResourceContentGetter.ts

```diff
@@ -14,8 +15,23 @@ export class CDTPResourceContentGetter extends CDTPEnableableDiagnosticsModule<C
         @inject(TYPES.CDTPClient)
         protected _protocolApi: CDTP.ProtocolApi,
         @inject(TYPES.IDomainsEnabler) domainsEnabler: CDTPDomainsEnabler,
+        @inject(TYPES.ExecutionContextEventsProvider) executionContextEventsProvider: IExecutionContextEventsProvider,
     ) {
         super(domainsEnabler);
+
+        executionContextEventsProvider.onExecutionContextCreated(executionContext => {
+            this.enable().then(() =>
+            /* The first time we call getResourceContent it reloads all resources in case any was evicted from the cache. If the debuggee is paused while that happens,
```

Comment:
Strange, is this an issue in v1 too?

Reply:
In v1 we have a different implementation. If we have a .html with 3 script tags we show 3 documents (So we don't need this API in v1).

On v2 instead we show a single document which is the .html web-page (So we need this API).